### PR TITLE
Refact2

### PR DESCRIPTION
**Pull request by rofl0r**

### CODE REVIEW

File: src/pixiewps.c [hunk 1]
```diff
@@ -293,19 +293,23 @@ unsigned int hardware_concurrency()
...
 	struct glibc_prng glibc_prng;
 
 	int32_t res;
 	int i = 0; /* Must hold MODE3_TRIES */
 	uint8_t tmp_s_nonce[16];
-
-	DEBUG_PRINT("Trying forward in time");
+	int break_cond = (MODE3_TRIES+1) * dir;
+	
+	if(dir == 1)
```
Comment (wiire-a) [inline src/pixiewps.c]: Missing a space.

File: src/pixiewps.c [hunk 2]
```diff
@@ -326,16 +330,21 @@ int find_rtl_es(struct global *wps, char* pin)
...
 				memcpy(wps->e_s1, wps->e_nonce, WPS_SECRET_NONCE_LEN); /* E-S1 = E-Nonce != E-S2 */
 				memcpy(tmp_s_nonce, wps->e_s2, WPS_SECRET_NONCE_LEN);  /* Chaching for next round, see below */
 			}
 			else {
 				memcpy(wps->e_s1, tmp_s_nonce, WPS_SECRET_NONCE_LEN);
 				memcpy(tmp_s_nonce, wps->e_s2, WPS_SECRET_NONCE_LEN);  /* E-S1 = old E-S1, E-S2 = new E-S2 */
 			}
-			wps->s1_seed = wps->nonce_seed + i - 1;
-			wps->s2_seed = wps->nonce_seed + i;
+			if(dir == 1) {
```
Comment (wiire-a) [inline src/pixiewps.c]: Missing a space.

File: src/pixiewps.c [hunk 3]
```diff
@@ -326,16 +330,21 @@ int find_rtl_es(struct global *wps, char* pin)
...
+				wps->s1_seed = wps->nonce_seed + i - dir;
+				wps->s2_seed = wps->nonce_seed + i;
+			} else {
```
Comment (wiire-a) [inline src/pixiewps.c]: The keyword should be on a new line:
```C
}
else {
```

File: src/pixiewps.c [hunk 4]
```diff
@@ -293,19 +293,23 @@ unsigned int hardware_concurrency()
 #endif
 }
 
-int find_rtl_es(struct global *wps, char* pin)
+static int find_rtl_es_dir(struct global *wps, char* pin, int dir)
```
Comment (wiire-a) [inline src/pixiewps.c]: The asterisk should be next to the variable, not next to the type (I missed this in your previous commit):
```C
char *pin
```

File: src/pixiewps.c [hunk 5]
```diff
@@ -293,19 +293,23 @@ unsigned int hardware_concurrency()
...
 {
 	uint_fast8_t found_p_mode = NONE;
 	struct glibc_prng glibc_prng;
 
 	int32_t res;
 	int i = 0; /* Must hold MODE3_TRIES */
 	uint8_t tmp_s_nonce[16];
-
-	DEBUG_PRINT("Trying forward in time");
+	int break_cond = (MODE3_TRIES+1) * dir;
```
Comment (wiire-a) [inline src/pixiewps.c]: Add spacing between arithmetical operations:
```C
(MODE3_TRIES + 1)
```

File: src/pixiewps.c [hunk 6]
```diff
@@ -354,63 +363,21 @@ int find_rtl_es(struct global *wps, char* pin)
...
-				wps->s1_seed = wps->nonce_seed - i;
-				wps->s2_seed = wps->nonce_seed - i + 1;
+	return found_p_mode;
+}
 
-				DEBUG_PRINT("Trying (%10u) with E-S1: ", wps->s1_seed);
-				DEBUG_PRINT_ARRAY(wps->e_s1, WPS_SECRET_NONCE_LEN);
-				DEBUG_PRINT("Trying (%10u) with E-S2: ", wps->s2_seed);
-				DEBUG_PRINT_ARRAY(wps->e_s2, WPS_SECRET_NONCE_LEN);
+static int find_rtl_es(struct global *wps, char* pin)
```
Comment (wiire-a) [inline src/pixiewps.c]: Asterisk next to the variable not the type.

**Comment by wiire-a** [comment]
I want to point out that when #63 came out I just copy and pasted the old code for a quick fix, without thinking of refactoring.

The same is true for many other parts of code in this repository: "features" have been building up without a proper refactoring (e.g. the `crack()` function should be rewritten). So yeah there is a lot of duplicate code :)

There are also a lot of special cases depending on the input and mode selected (e.g. `mode 3`, `--m7-enc`, `-S` etc.).

**Comment by rofl0r** [comment]
updated

**Pull request merged**